app/services/course/assessment/question/programming_codaveri/java/java_package_service.rb

```diff
@@ -129,10 +129,10 @@ def preload_question_test_cases
   end
 
   def extract_pathname_from_file(file_content, pathname)
-    class_name_extractor = /public\s+class\s+([A-Za-z_][A-Za-z0-9_]*)/
+    class_name_extractor = /(^|;)\s*public\s+class\s+([A-Za-z_][A-Za-z0-9_]*)/
```

Comment:
duplicated function, can we move it to a common location?  Otherwise we'll risk overlooking updating some in future.